plugindocs.rb

```diff
@@ -130,6 +132,27 @@ def tag(version)
     version ? "v#{version}" : "main"
   end
 
+  # returns alias mappings for each plugin type & name ([type][target]=alias) ex: (["input"]["beats"]="agent")
+  def load_alias_mappings
+    alias_util = Util::AliasPluginUtil.new
+    yaml = alias_util.fetch_alias_mappings
+
+    aliases = Hash.new
+
+    yaml.each do |type, alias_defs|
+      aliases[type] = Hash.new
+      alias_defs.each do |alias_definition|
+        mapping = {}
+        mapping["alias"] = alias_definition["alias"]
+        mapping["doc_headers"] = alias_definition["docs"]
+        aliases[type][alias_definition["from"]] ||= []
+        aliases[type][alias_definition["from"]] << mapping
+      end
+    end
+
+    aliases
```

Comment:
We have a fair bit of extra logic in here that is required by our structuring the intermediate index of aliases by type first and then by name, that could be obviated if we instead pre-indexed them by the plugin's fully qualified name:

```suggestion
    yaml.each_with_object(Hash.new { |h,k| h[k]=[] }) do |aliases, (type, alias_defs)|
      alias_defs.each do |alias_definition|
        canonical_name = alias_definition.fetch("from")
        aliases["#{type}-#{canonical_name}"] << {
          "type" => type,
          "from" => canonical_name,
          "alias" => alias_definition.fetch("alias"),
          "docs_headers" => alias_definition.fetch("docs", {}),
        }
      end
    end
```

Propagating the `type` and `from` into the indexed definitions ensures that they are available in the VPR's use where they are needed.

This of course creates a more flat mapping, which we would query directly (see other comments with `FLAT_ACCESS`)

Reply:
Simplified the logic, now we don't necessarily created another structure form domain YAML Hash structure.